### PR TITLE
localization scripts

### PR DESCRIPTION
**Pull request by adityamulik**
Hello,

The pull request is for the purpose of automating the localization of galaxy_ng. 
We're actively developing an Ansible collection for this process called as [Ansible Memsource Collection](https://github.com/ansible/ansible-collection-memsource)

The process comprises of two tasks:
- Pull Strings from the project (Using the project framework's strings extraction command). These strings (Usually retrieved are in English) are then uploaded to a third party software called Memsource
- Push Strings to the projects, once the translations are completed and the strings are ready to be pushed back to the repository, the push/ post_translation role of this collection is executed

For this process, we require two shell scripts to be placed in the repository. 
- pre_translation.sh - which performs the extraction of strings from a project. In this process, a folder called "translations" will be provided, the script ensures to move the strings to the translations folder
   - Empty translations folder -> django.po file will be dropped here which will be uploaded to memsource.
- post_translation.sh - which performs the process of moving the strings to appropriate directory in the project, a PR is later autogenerated by the underlying Ansible role being utilized here for merging the translated strings. Again a translations folder will be provided with all the translated strings, the script would ensure to utilize the folder and move the files according to the project needs.
   - Translation folder with all languages generated files
   e.g. 
          /translations/fr/django.po 
          /translations/jp/django.po
          /translations/ko/django.po

  The script would modify the file 
  e.g.
        /translations/fr/django.po -> /translations/fr/LC_MESSAGES/django.po -> Moving to galaxy_ng/locale/fr/LC_MESSAGES/django.po

### CODE REVIEW

**Comment by himdel** [comment]
@adityamulik there seems to be a rebase issue here, this is now changing 68 files

(hint: never ever use `git merge` with feature branches, this probably happened because you tried to rebase a merge commit :))

(`git reset --hard 36ee12837b9ba4af414cd2db0905cf3c520b6721; git rebase upstream/master` should probably do the trick)

**Comment by himdel** [comment]
(otherwise, same issues as https://github.com/ansible/ansible-hub-ui/pull/2309)

**Comment by adityamulik** [comment]
> @adityamulik there seems to be a rebase issue here, this is now changing 68 files
> 
> (hint: never ever use `git merge` (or `git pull` without `--ff-only`) with feature branches, this probably happened because you tried to rebase a merge commit :))
> 
> (`git reset --hard 36ee12837b9ba4af414cd2db0905cf3c520b6721; git rebase upstream/master` should probably do the trick)

Much thanks for the inputs, it worked as expected :)

**Comment by rooftopcellist** [comment]
@himdel is there anything blocking merging this?

File: MANIFEST.in [hunk 1]
```diff
@@ -5,6 +5,8 @@ include integration_requirements.txt
 include unittest_requirements.txt
 include pyproject.toml
 include requirements/requirements.common.txt
+include tools/scripts/l18n/post_translation.sh
+include tools/scripts/l18n/pre_translation.sh
```
Comment (himdel) [inline MANIFEST.in:9]: this is surprising .. we should not distribute these scripts as part of the resulting pypi package

**Comment by rooftopcellist** [comment]
The idea was to keep the memsource collection generic and have all logic specific to the upstream repos live with each project.  But those are valid points.  I opened up an issue to add support for specifying the path for a script like this in the collection, that way we don't have to vendor it in the upstream app repo.  

https://github.com/ansible/ansible-collection-memsource/issues/39

**Comment by rooftopcellist** [comment]
I've added the ability to specify a script local to the memsource roles which gets copied into the clone of the app (in this case ansible-hub-ui), so we can close this PR and move forward with that.  
* https://github.com/ansible/ansible-collection-memsource/pull/45

**Comment by adityamulik** [comment]
Closing the PR as the requirement of local translation scripts have been added as per pr https://github.com/ansible/ansible-collection-memsource/pull/45

**Pull request closed without merging**